packages/ui/src/components/card-select.tsx

```diff
@@ -0,0 +1,184 @@
+import { createContext, forwardRef, HTMLAttributes, ReactNode, useContext, useState } from 'react'
+
+import { cn } from '@utils/cn'
+import { cva } from 'class-variance-authority'
+
+import { Icon } from './icon'
+import { Logo } from './logo'
+
+type CardSelectType = 'single' | 'multiple'
+
+interface CardSelectRootProps<T> {
+  type: CardSelectType
+  name?: string
+  value?: T extends 'single' ? unknown : unknown[]
+  defaultValue?: T extends 'single' ? unknown : unknown[]
+  onValueChange?: T extends 'single' ? (val: unknown) => void : (val: unknown[]) => void
+  disabled?: boolean
+  orientation?: 'horizontal' | 'vertical'
+  gap?: 'sm' | 'md' | 'lg'
+  children: ReactNode
+}
+
+interface CardSelectItemProps extends HTMLAttributes<HTMLInputElement> {
+  value: unknown
+  icon?: React.ComponentProps<typeof Icon>['name']
+  logo?: React.ComponentProps<typeof Logo>['name']
+  disabled?: boolean
+  children: ReactNode
+}
+
+interface CardSelectContext {
+  type: CardSelectType
+  name: string
+  currentValue: unknown | unknown[]
+  disabled: boolean
+  onValueChange: (value: unknown) => void
+}
+
+const CardSelectContext = createContext<CardSelectContext | null>(null)
+
+function useCardSelect() {
+  const context = useContext(CardSelectContext)
+  if (!context) {
+    throw new Error('CardSelect.Item must be used within CardSelect.Root')
+  }
+  return context
+}
+
+function isChecked(value: unknown, current: unknown | unknown[]) {
+  return Array.isArray(current) ? current.includes(value) : current === value
+}
+
+const cardSelectVariants = cva('cn-card-select-root', {
+  variants: {
+    orientation: {
+      vertical: '',
+      horizontal: 'cn-card-select-horizontal'
+    },
+    gap: {
+      sm: 'cn-card-select-gap-sm',
+      md: 'cn-card-select-gap-md',
+      lg: 'cn-card-select-gap-lg'
+    }
+  },
+  defaultVariants: {
+    orientation: 'vertical',
+    gap: 'md'
+  }
+})
+
+function CardSelectRoot<T extends CardSelectType>({
+  type,
+  orientation = 'vertical',
+  gap = 'md',
+  name = `card-select-${Math.random().toString(36).slice(2)}`,
```

Comment:
we added a uuid util for another component recently.. let's reuse that, along with a `useMemo` to avoid the name changing on rerender